### PR TITLE
Support stylelintrc

### PR DESCRIPTION
**Pull request by SevenOutman**
## Description

Treat filename `.stylelintrc` (without any extension) [as YAML/JSON format](https://stylelint.io/user-guide/configure#configuration). 

Related issue #10920 

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by SevenOutman** [comment]
Not clear how to describe this change following `changelog_unreleased/TEMPLATE.md`.

File: src/language-yaml/embed.js [hunk 1]
```diff
@@ -7,7 +7,8 @@ function embed(path, print, textToDoc, options) {
   if (
     node.type === "root" &&
     options.filepath &&
-    /(?:[/\\]|^)\.prettierrc$/.test(options.filepath)
+    (/(?:[/\\]|^)\.prettierrc$/.test(options.filepath) ||
```
Comment (fisker) [inline src/language-yaml/embed.js:10]:
```suggestion

File: src/language-yaml/embed.js [hunk 2]
```diff
@@ -7,7 +7,8 @@ function embed(path, print, textToDoc, options) {
...
+      /(?:[/\\]|^)\.stylelintrc$/.test(options.filepath))
```
Comment (fisker) [inline src/language-yaml/embed.js:11]:
```suggestion
    /(?:[/\\]|^)\.(?:prettier|stylelint)rc$/.test(options.filepath)
```

File: tests/integration/__tests__/with-parser-inference.js [hunk 3]
```diff
@@ -28,6 +28,18 @@ describe("infers parser from filename", () => {
     ).toEqual("{}\n");
   });
 
+  test("json from .stylelintrc", () => {
+    expect(
```
Comment (fisker) [inline tests/integration/__tests__/with-parser-inference.js:32]: We do format test in `tests/format` dir, see https://github.com/prettier/prettier/pull/8105/files

**Comment by thorn0** [comment]
@SevenOutman 

> Not clear how to describe this change following `changelog_unreleased/TEMPLATE.md`.

A title and a short description will suffice. No need to add a code example.

File: tests/integration/__tests__/with-parser-inference.js (hunk 3 above)
Comment (SevenOutman) [reply tests/integration/__tests__/with-parser-inference.js:32]: Isn't this suite for testing infer file format from filename with not extensions?

Comment (fisker) [reply tests/integration/__tests__/with-parser-inference.js:32]: Okay, works too.

**Review by fisker — APPROVED** [review]

**Review by sosukesuzuki — APPROVED** [review]

**Pull request merged**